mrbgems/picoruby-watchdog/ports/esp32/watchdog.c

```diff
@@ -0,0 +1,57 @@
+#include "esp_task_wdt.h"
+
+void
+Watchdog_enable(uint32_t delay_ms, bool pause_on_debug)
+{
+  esp_task_wdt_config_t twdt_config = {
+    .timeout_ms = delay_ms,
+    .trigger_panic = pause_on_debug,
+  };
+  esp_task_wdt_init(&twdt_config);
+  esp_task_wdt_add(NULL);
+}
+
+void
+Watchdog_disable(void)
+{
+  esp_task_wdt_deinit();
+}
+
+void
+Watchdog_reboot(uint32_t delay_ms)
+{
+  /* Not implemented */
+}
+
+void
+Watchdog_start_tick(uint32_t cycles)
+{
+  /* Not implemented */
+}
+
+void
+Watchdog_update(void)
+{
+  esp_task_wdt_reset();
+}
+
+bool
+Watchdog_caused_reboot(void)
+{
+  /* Not implemented */
+  return false;
```

Comment:
If this function returns false, users are going to misunderstand the reboot reason.
Even if you don't want to implement the feature, returning false would not be a good way.
I wonder why don't you implement it? I think you can do it with `esp_reset_reason()`, can't you?

Reply:
Sorry, I was not aware of the existence of esp_reset_reason().
I used this function to implement it.